src/test/java/com/yelp/nrtsearch/server/luceneserver/facet/RuntimeScriptFacetsTest.java

```diff
@@ -0,0 +1,374 @@
+            nums.add("2");
+            return nums;
+          }
+        }
+
+        public static class DocValueScript extends RuntimeScript {
+
+          public DocValueScript(
+              Map<String, Object> params, DocLookup docLookup, LeafReaderContext leafContext) {
+            super(params, docLookup, leafContext);
+          }
+
+          @Override
+          public Object execute() {
+            Map<String, LoadedDocValues<?>> doc = getDoc();
+            return doc.get("atom_1").get(0) + "_" + doc.get("atom_2").get(0);
+          }
+        }
+      }
+    }
+  }
+
+  protected List<String> getIndices() {
+    return Collections.singletonList(DEFAULT_TEST_INDEX);
+  }
+
+  protected FieldDefRequest getIndexDef(String name) throws IOException {
+    return getFieldsFromResourceFile("/facet/facet_script_facets.json");
+  }
+
+  protected void initIndex(String name) throws Exception {
+    IndexWriter writer = getGlobalState().getIndex(name).getShard(0).writer;
+    // don't want any merges for these tests
+    writer.getConfig().setMergePolicy(NoMergePolicy.INSTANCE);
+
+    // add documents one chunk at a time to ensure multiple index segments
+    List<AddDocumentRequest> requestChunk = new ArrayList<>();
+    for (int id = 0; id < NUM_DOCS; ++id) {
+      requestChunk.add(
+          AddDocumentRequest.newBuilder()
+              .setIndexName(name)
+              .putFields(
+                  "doc_id",
+                  AddDocumentRequest.MultiValuedField.newBuilder()
+                      .addValue(String.valueOf(id))
+                      .build())
+              .putFields(
+                  "int_field",
+                  AddDocumentRequest.MultiValuedField.newBuilder()
+                      .addValue(String.valueOf(id))
+                      .build())
+              .putFields(
+                  "atom_1",
+                  AddDocumentRequest.MultiValuedField.newBuilder()
+                      .addValue(String.valueOf(id % 3))
+                      .build())
+              .putFields(
+                  "atom_2",
+                  AddDocumentRequest.MultiValuedField.newBuilder()
+                      .addValue(String.valueOf(id % 2))
+                      .build())
+              .build());
+
+      if (requestChunk.size() == SEGMENT_CHUNK) {
+        addDocuments(requestChunk.stream());
+        requestChunk.clear();
+        writer.commit();
+      }
+    }
+  }
+
+  @Override
+  protected List<Plugin> getPlugins(LuceneServerConfiguration configuration) {
+    return Collections.singletonList(new TestRuntimeScriptPlugin());
+  }
+
+  @Test
+  public void RuntimeScriptForInt() {
+
+    RuntimeField runtimeField =
+        RuntimeField.newBuilder()
+            .setScript(Script.newBuilder().setLang("painless").setSource("int").build())
+            .setName("runtime_field")
+            .build();
+
+    List expectedValues = new ArrayList<>();
+    for (int id = 0; id < SEGMENT_CHUNK; ++id) {
```

Comment:
I don't think this is based on the `SEGMENT_CHUNK`, but the hard coded top hits in the query. You probably want to create a new `TOP_HITS` constant to avoid confusion.